libsysinfo/disk.py

```diff
@@ -27,47 +29,55 @@ def _get_mounts(mounts_file="/proc/mounts"):
         total_space = (stats.f_blocks * block_size) / megabytes
         free_space = (stats.f_bfree * block_size) / megabytes
 
-        yield {"device": device,
-               "mount-point": mount_point,
-               "filesystem": filesystem,
-               "total-space": int(total_space),
-               "free-space": int(free_space)}
+        yield {
+            "device": device,
+            "mount-point": mount_point,
+            "filesystem": filesystem,
+            "total-space": int(total_space),
+            "free-space": int(free_space),
+        }
 
 
-def _get_filesystem_for_path(path, mounts_file="/proc/mounts"):
+def _get_filesystem_for_path(
+    path: str, mounts_file: str = "/proc/mounts"
+) -> dict[str, str | int] | None:
     candidate = None
 
     path = os.path.realpath(path)
     path_segments = path.split("/")
 
     for info in _get_mounts(mounts_file):
-        if not path.startswith(info["mount-point"]):
+        # info["mount-point"] is a str
+        if not path.startswith(str(info["mount-point"])):
             continue
 
-        mount_segments = info["mount-point"].split("/")
+        mount_segments = str(info["mount-point"]).split("/")
 
-        if ((not candidate) or
-                path_segments[:len(mount_segments)] == mount_segments):
+        if (not candidate) or path_segments[
+            : len(mount_segments)
+        ] == mount_segments:
```

Comment:
I'm normally a big fan of splitting long stuff over lines. But splitting a slice makes this seem very strange.

Reply:
Yeah, I agree - that was what ruff did?!

Reply:
I guess so. Might be an option to disable slice splitting, although I'm not sure.